extensions/amp-form/0.1/amp-form.js

```diff
@@ -16,14 +16,90 @@
 
 import {isExperimentOn} from '../../../src/experiments';
 import {getService} from '../../../src/service';
+import {assertHttpsUrl} from '../../../src/url';
+import {user} from '../../../src/log';
+import {onDocumentReady} from '../../../src/document-ready';
+import {xhrFor} from '../../../src/xhr';
+import {toArray} from '../../../src/types';
+import {startsWith} from '../../../src/string';
+
 
 /** @type {string} */
 const TAG = 'amp-form';
 
+export class AmpForm {
+
+  /**
+   * Adds functionality to the passed form element and listens to submit event.
+   * @param {!HTMLFormElement} element
+   */
+  constructor(element) {
+    /** @const @private {!Window} */
+    this.win_ = element.ownerDocument.defaultView;
+
+    /** @const @private {!Element} */
+    this.form_ = element;
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
+
+    /** @const @private {string} */
+    this.method_ = this.form_.getAttribute('method') || 'GET';
+
+    /** @const @private {?string} */
+    this.xhrAction_ = this.form_.getAttribute('action-xhr');
+    if (this.xhrAction_) {
+      assertHttpsUrl(this.xhrAction_, this.form_, 'action-xhr');
```

Comment:
Before doing these checks, we need to resolve `xhrAction_` to the full URL. Otherwise we can't test if it starts with `https://...` is not valid.